src/wwwroot/services/auth.js

```diff
@@ -157,73 +182,57 @@
     }
 
     function getApiToken() {
-      if (temporarilyAuthed) {
-        return encodedToken;
-      } else {
-        ensureToken();
-        return encodedToken;
+      if (!loginSession) {
+        return null;
+      }
+      if (loginSession.temporaryToken) {
+        return loginSession.token
       }
+      ensureToken();
+      return loginSession && loginSession.token;
     }
-
+    
     function ensureToken() {
-      // It is to avoid have different users in UI and AJAX requests
-      // TODO: improve it because when token is removed, the request is done anyway, maybe it is possible to cancel request when there is no token
```

Comment:
What do you think about resurrecting these comments? 

I think that the _TODO note_ is still relevant, and we could clarify a little more the explanation in the first line, what do you think about:

```
// It is to allow apply logout in all open tabs and also to avoid having different users in UI and AJAX requests.
```

Reply:
I will add them. thx